### PR TITLE
SRX-AEY0CN Datadog Metrics

### PR DESCRIPTION
**Pull request by leandro-freiheit**
Old PR: https://github.com/freiheit-com/kuberpult/pull/367

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -124,6 +124,23 @@ func UpdateDatadogMetrics(state *State) error {
 	return nil
 }
 
+func SendRegularlyDatadogMetrics(repo Repository, interval int, callBack func(repository Repository)) {
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:127]:
```suggestion
func SendRegularlyDatadogMetrics(repo Repository, interval int, callBack func(repository Repository)) {
```
either make the `interval` a `Duration` type, or rename it to `intervalSeconds`.

File: services/cd-service/pkg/cmd/server.go [hunk 2]
```diff
@@ -70,6 +71,8 @@ func RunServer() {
 
 		var c Config
 
+		var wg sync.WaitGroup
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/cmd/server.go:74]: what's the purpose of the waitgroup?

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:127]: Either
```suggestion
func RegularlySendDatadogMetrics(repo Repository, interval int, callBack func(repository Repository)) {
```
OR:
```suggestion
func SendDatadogMetricsRegularly(repo Repository, interval int, callBack func(repository Repository)) {
```

File: services/cd-service/pkg/cmd/server.go (hunk 2 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/cmd/server.go:74]: I think we don't need it.

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (leandro-freiheit) [reply services/cd-service/pkg/repository/transformer.go:127]: Fixed, changed interval to a Duration

Comment (leandro-freiheit) [reply services/cd-service/pkg/repository/transformer.go:127]: Fixed, function is now RegularlySendDatadogMetrics

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**